osf/models/user.py

```diff
@@ -576,7 +576,7 @@ def visible_contributor_to(self):
         """
         Nodes where user is a bibliographic contributor (group membership not factored in)
         """
-        return self.nodes.filter(is_deleted=False, contributor__visible=True, type__in=['osf.node', 'osf.registration'])
+        return self.nodes.filter(is_deleted=False, contributor__visible=True, contributor__user=self, type__in=['osf.node', 'osf.registration'])
```

Comment:
The fix make sense since we need to make sure that the invisible contributor on the project must be the `self` user.

Without it, `visible_contributor_to` returns two projects (in unit tests) with the following contributor set for each.

```
<QuerySet [<Contributor(user=freddiemercury+8522846@cos.io, visible=True, permission=admin)>]>

<QuerySet [<Contributor(user=freddiemercury+8822275@cos.io, visible=True, permission=admin)>, <Contributor(user=freddiemercury+8522846@cos.io, visible=False, permission=write)>]>
```

Apparently, if we only have `contributor__visible=True`, the query find nodes as long as one of the contributor is visible. I am not sure why this happens after the upgrade, might be an issue with the related manager.

```bash
In [7]: user.nodes
Out[7]: <django.db.models.fields.related_descriptors.create_forward_many_to_many_manager.<locals>.ManyRelatedManager at 0x7fcbb4a19898>
```

Reply:
Update/Example:

We have user `7773344` and `8079719`; we have Project `9qnbu` and `x8jc3`
Project  `9qnbu` has contributor `7773344` as visible
Project `x8jc3` has contributor `7773344` & `8079719` as visible

```
return self.nodes.filter(is_deleted=False, contributor__visible=True, type__in=['osf.node', 'osf.registration'])
```

The above returns 3 items instead of 2. Apparently, `contributor__visible=True` is looped on each contributor.

```
<AbstractNodeQuerySet [
    (title='Visionary radical matrix', category='project') with guid 'x8jc3',
    (title='Adaptive modular access', category='project') with guid '9qnbu',
    (title='Adaptive modular access', category='project') with guid '9qnbu'
]>
```

My concern is, even with `contributor__user=self`, it still loops each contributor on every project ... this makes sense why adding it fix the tests but it might be too costly?